
version: '2.1'

volumes:
  pg-backup-data-dir:
  pg-data-dir:

services:

  db:
    image: kartoza/postgis:14-3.1
    restart: 'always'
    volumes:
      - ./setup-db.sql:/docker-entrypoint-initdb.d/setup-db.sql
    environment:
      - POSTGRES_DB=gis
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
    healthcheck:
      interval: 60s
      timeout: 30s
      retries: 3
      test: "pg_isready"

  pg_restore:
    image: kartoza/pg-backup:manual-build
    restart: 'always'
    volumes:
      - pg-backup-data-dir:/backups
      - ./tests:/tests
      - ../utils:/lib/utils
    environment:
      - DUMPPREFIX=PG_gis
      - POSTGRES_HOST=db
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
      - POSTGRES_PORT=5432
      - TARGET_DB=gis
      - WITH_POSTGIS=1
      - STORAGE_BACKEND=S3
      - ACCESS_KEY_ID=${ACCESS_KEY_ID}
      - SECRET_ACCESS_KEY=${SECRET_ACCESS_KEY}
      - DEFAULT_REGION=${DEFAULT_REGION}
      - BUCKET=${BUCKET}
      - HOST_BASE=${HOST_BASE}
      - HOST_BUCKET=${HOST_BUCKET}
      - SSL_SECURE=${SSL_SECURE}
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      interval: 60s
      timeout: 30s
      retries: 3
      test: "pg_isready"
